drv/lpc55-swd/build.rs

```diff
@@ -88,6 +97,163 @@ fn generate_swd_functions(config: &TaskConfig) -> Result<()> {
     Ok(())
 }
 
+fn prepare_endoscope() -> Result<(), anyhow::Error> {
+    let key = "CARGO_BIN_FILE_ENDOSCOPE";
+    let elf_path = PathBuf::from(
+        std::env::var(key).context(format!("Cannot read env var '{}'", key))?,
+    );
+    let data = std::fs::read(&elf_path).context("could not open ELF file")?;
+    let elf = Elf::parse(&data).context("cannot parse ELF file")?;
+
+    let out_dir = build_util::out_dir();
+    let dest_path = out_dir.join("endoscope.rs");
+    let bin_path = out_dir.join("endoscope.bin");
+    let mut file = std::fs::File::create(&dest_path)
+        .context("Cannot create output file")?;
+
+    writeln!(&mut file, "mod endoscope {{")?;
+
+    // Get entry point
+    if elf.header.container()? != Container::Little
+        || elf.header.e_machine != goblin::elf::header::EM_ARM
+    {
+        bail!("Not a little endian ARM ELF file");
+    }
+
+    let mut interesting = std::collections::BTreeMap::new();
+    for name in [
+        // Load address of image
+        (LOAD_SYMBOL, "LOAD"),
+        // Address of endoscope output struct
+        (SHARED_STRUCT_SYMBOL, "SHARED"),
+    ]
+    .iter()
+    {
+        interesting.insert(name.0, name.1);
+    }
+
+    for sym in elf.syms.iter() {
+        if let Some(str) = elf.strtab.get_at(sym.st_name) {
+            let name = format!("{:#}", demangle(str));
+            if let Some(myname) = interesting.get(name.as_str()) {
+                writeln!(
+                    &mut file,
+                    "pub const {}:u32 = 0x{:x};",
```

Comment:
Formatting nit:
```suggestion
                    "pub const {}: u32 = {:#x};",
```
(I know we run `rustfmt` later, but this was confusing on first glance)

I also changed the second format string, because you can get the leading `0x` automatically with the `#` directive.